packages/connection/src/class-owner.php

```diff
@@ -0,0 +1,184 @@
+<?php
+/**
+ * The Connection\Owner class file.
+ *
+ * @package automattic/jetpack-connection
+ */
+
+namespace Automattic\Jetpack\Connection;
+
+use Automattic\Jetpack\Redirect;
+use Automattic\Jetpack\Tracking;
+
+/**
+ * The Connection\Owner class.
+ */
+class Owner {
+
+	/**
+	 * This is an entire admin notice dedicated to messaging and handling of the case where a user is trying to delete
+	 * the connection owner.
+	 */
+	public function delete_user_update_connection_owner_notice() {
+		global $current_screen;
+
+		/*
+		 * phpcs:disable WordPress.Security.NonceVerification.Recommended
+		 *
+		 * This function is firing within wp-admin and checks (below) if it is in the midst of a deletion on the users
+		 * page. Nonce will be already checked by WordPress, so we do not need to check ourselves.
+		 */
+
+		if ( ! isset( $current_screen->base ) || 'users' !== $current_screen->base ) {
+			return;
+		}
+
+		if ( ! isset( $_REQUEST['action'] ) || 'delete' !== $_REQUEST['action'] ) {
+			return;
+		}
+
+		// Get connection owner or bail.
+		$connection_manager  = new Manager();
+		$connection_owner_id = $connection_manager->get_connection_owner_id();
+		if ( ! $connection_owner_id ) {
+			return;
+		}
+		$connection_owner_userdata = get_userdata( $connection_owner_id );
+
+		// Bail if we're not trying to delete connection owner.
+		$user_ids_to_delete = array();
+		if ( isset( $_REQUEST['users'] ) ) {
+			$user_ids_to_delete = array_map( 'sanitize_text_field', wp_unslash( $_REQUEST['users'] ) );
+		} elseif ( isset( $_REQUEST['user'] ) ) {
+			$user_ids_to_delete[] = sanitize_text_field( wp_unslash( $_REQUEST['user'] ) );
+		}
+
+		// phpcs:enable
+		$user_ids_to_delete        = array_map( 'absint', $user_ids_to_delete );
+		$deleting_connection_owner = in_array( $connection_owner_id, (array) $user_ids_to_delete, true );
+		if ( ! $deleting_connection_owner ) {
+			return;
+		}
+
+		// Bail if they're trying to delete themselves to avoid confusion.
+		if ( get_current_user_id() === $connection_owner_id ) {
+			return;
+		}
+
+		// Track it!
+		$tracking = new Tracking();
+		if ( method_exists( $tracking, 'record_user_event' ) ) {
```

Comment:
I checked the [PR](https://github.com/Automattic/jetpack/pull/13388) where this check was added, and it looks like the intention was just to be overly defensive. Since the `Automattic\Jetpack\Tracking` class has always had the `record_user_event` method, I removed this `method_exists` check in commit e6540b8.